src/common.c

```diff
@@ -514,8 +514,31 @@ tree_t find_element_mode_indication(tree_t view, tree_t field, bool *converse)
          const int nelems = type_fields(view_type);
          for (int i = 0; i < nelems; i++) {
             tree_t e = type_field(view_type, i);
-            if (tree_ref(e) == field)
+            if (tree_ref(e) == field) {
+               // If this element has a view and we need to apply converse,
+               // recursively apply converse to the element's view
+               if (*converse && tree_has_value(e)) {
```

Comment:
The idea is the caller checks `*converse` or passes it to `converse_mode()`. Of course it's possible that some callers handle it incorrectly.